Orm/Xtensive.Orm/Orm/Rse/Transformation/Internals/SkipTakeRewriterState.cs

```diff
@@ -56,12 +54,25 @@ private static Func<int> PositiveSelector(Func<int> valueSelector)
       };
     }
 
-    public IDisposable CreateScope()
+    internal readonly ref struct SkipTakeRewriterScope
+    {
+      private readonly SkipTakeRewriter rewriter;
+      private readonly SkipTakeRewriterState prevState;
+
+      public void Dispose() => rewriter.State = prevState;
+
+      public SkipTakeRewriterScope(SkipTakeRewriter rewriter, SkipTakeRewriterState prevState)
+      {
+        this.rewriter = rewriter;
+        this.prevState = prevState;
+      }
+    }

```

Comment:
This nested ```struct``` declaration should also be moved to the top of the class.

Reply:
Done.